### PR TITLE
Update to payjoin-0.21

### LINKED ISSUE #736: Test payjoin_ffi backwards compatibility with v1
v1-to-v1 is no longer the most relevant payjoin-ffi since it doesn't support payjoin-ffi doesn't ship a pure v1 receiver. The following are the relevant code paths to test:

- [x] v2-to-v2
- [ ] v1-to-v2
- [ ] v2-to-v1 (with `rust-payjoin::v1` feature, manual `extract_v1` would ignore handling the v1 receiver version error)

Testing each of these should ensure that v1-to-v1 actually works, but will never be hit with our shipped state machines.

### PR DESCRIPTION
**Pull request by DanGould**
- 1st commit simplifies lib.rs with pub use and wildcard exports and builds with flags properly
- 2nd commit matches feature name danger-local-https
- 3rd commit creates new modules to more closely match the rust-payjoin source it wraps
- 4th commit binds to 0.21 prerelease including passing integration tests

PYTHON INTEGRATION IS NOW WRONG! V1-to-V1 tests ARE NOW WRONG / USELESS, and were removed. these must be FIXED: payjoin/rust-payjoin#736. I have deleted the v1-to-v1 tests but they may be revived to satisfy v2.

Python is being ignored since our immediate goal is to get payjoin-flutter working

### CODE REVIEW

**Review by spacebear21 — APPROVED** [review]
I don't know much about FFI but this looks fairly straightforward.

Out of curiosity, is it possible to use [bitcoin-ffi](https://github.com/bitcoindevkit/bitcoin-ffi) for the rust-bitcoin bindings instead of making our own?

**Comment by DanGould** [comment]
Yes it is possible to depend on `bitcoin-ffi` types (`TxIn`, `Txout`, `OutPoint`, `Network`) and upstream our `PsbtInput` type to that dependency. I'm hoping they release, but we can depend on a git commit from that repo.

edit: see payjoin/payjoin-ffi#3

File: README.md [hunk 1]
```diff
@@ -63,7 +63,7 @@ The integration tests illustrates and verify integration using bitcoin core and
 ```shell
 
 # Run the integration test
-cargo test --package payjoin_ffi --test bitcoin_core_integration_test v1_to_v1_full_cycle
+cargo test --package payjoin_ffi --test bitcoin_core_integration v1_to_v1_full_cycle
```
Comment (spacebear21) [inline README.md]: This test was deleted afaict

File: src/send/uni.rs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+
+impl From<SenderBuilder> for super::SenderBuilder {
+    fn from(value: SenderBuilder) -> Self {
+        value.0
+    }
+}
+
+#[uniffi::export]
+impl SenderBuilder {
+    //TODO: Replicate all functions like this & remove duplicate code
```
Comment (spacebear21) [inline src/send/uni.rs]: Should we create a follow-up issue for this?

Comment (DanGould) [reply src/send/uni.rs]: shoot I thought I removed this. The TODO item is actually completed and I think was copied over from the original to make separate ffi/uniffi fns

File: README.md [hunk 3]
```diff
@@ -33,39 +33,14 @@ pip install payjoin
...
-Now, run the integration tests:
-
-The integration tests illustrates and verify integration using bitcoin core and with bitcoin dev kit(bdk).
+The integration tests illustrates and verify integration using bitcoin core and bdk.
 
 ```shell
 
 # Run the integration test
-cargo test --package payjoin_ffi --test bitcoin_core_integration_test v1_to_v1_full_cycle
 cargo test --package payjoin_ffi --test bdk_integration_test v1_to_v1_full_cycle
```
Comment (spacebear21) [inline README.md:42]: sorry i missed this one in my earlier review but this test is also obsolete.

File: src/send/uni.rs (hunk 2 above)
Comment (DanGould) [reply src/send/uni.rs]: fixed

**Review by spacebear21 — APPROVED** [review]
utACK c96d28cc30d99271207b7226b71a6b5402adb9ac

File: README.md (hunk 1 above)
Comment (DanGould) [reply README.md]: fixed by removing much of this section of the readme since we spin up using `bitcoind` crate now

**Pull request merged**